Github/legos/github_get_open_branches/github_get_open_branches.py

```
@@ -0,0 +1,64 @@
+
+##
+##  Copyright (c) 2023 unSkript, Inc
+##  All rights reserved.
+##
+from pydantic import BaseModel, Field
+from typing import List
+import pprint
+from github import GithubException
+
+
+class InputSchema(BaseModel):
+    owner: str = Field(
+        description='Username of the GitHub user. Eg: "johnwick"', 
+        title='Owner'
+    )
+    repository: str = Field(
+        description='Name of the GitHub repository. Eg: "Awesome-CloudOps-Automation"',
+        title='Repository',
+    )
+
+
+def github_get_open_branches_printer(output):
+    if not output:
+        return
+    pprint.pprint(output)
+
+
+def github_get_open_branches(handle, owner: str, repository: str)-> List:
+    """github_get_open_branches returns 100 open github branches for a user.
+
+        :type handle: object
+        :param handle: Object returned from task.validate(...).
+    
+        :type owner: string
+        :param owner: Username of the GitHub user. Eg: "johnwick"
+
+        :type repository: string
+        :param repository: Name of the GitHub repository. Eg: "Awesome-CloudOps-Automation"
+
+        :rtype: List of branches for a user for a repository
+    """
+    result = []
+    try:
+        user = handle.get_user(login=owner)
+        repos = user.get_repos()
+        repo_name = owner+"/"+repository
+        if len(list(repos)) == 0:
+            return [f"{owner} does not have any repositories"]
+        for repo in repos:
+            if repo.full_name == repo_name:
+                branches = repo.get_branches()
+                [result.append(branch.name) for branch in branches[:100]]
+    except GithubException as e:
+        if e.status == 403:
+            return [f"You need push access"]
+        elif e.status == 404:
+            return [f"No such username or repository"]
+        else:
+            return [e.data]
```

@shloka-bhalgat-unskript we should raise the exception instead of return.